ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_service.rs

```diff
@@ -294,6 +323,50 @@ impl TableInfoService {
         Ok(())
     }
 
+    /// Tries to upload a snapshot of the database if the backup service is enabled.
+    /// This function is called to periodically back up the database state to Google Cloud Storage (GCS).
+    /// It checks the latest epoch of data already backed up in GCS and compares it with the current epoch.
+    async fn try_backup_db_snapshot_for_new_epoch(
```

Comment:
What's the flow of restoring this data? Should we only backup it at epoch ending?

Reply:
1. for restoring, usually for an empty node, it'll read the metadata in the GCS to determine the latest epoch and fetch the tar file generated for that epoch. After this, the db will be updated by the node. 
2. Yea; we internalize the check in this call. Updated the func name